application/src/main/java/org/togetherjava/tjbot/features/reminder/RemindRoutine.java

```diff
@@ -137,30 +139,29 @@ static ReminderRoute toPrivate(PrivateChannel channel) {
         }
     }
 
-    private void attemptRetryReminder(Reminder reminder, Throwable failure) {
-        if (reminder.pendingReminders().getFailureAttempts() > MAX_FAILURE_RETRY) {
+    private void attemptRetryReminder(PendingRemindersRecord pendingReminder, Throwable failure) {
+        if (pendingReminder.getFailureAttempts() > MAX_FAILURE_RETRY) {
             logger
                 .warn("""
                         Failed to send a reminder with (authorID '{}') skipping it. This can be due to a network issue, \
                         but also happen if the bot disconnected from the target guild and the \
-                        user has disabled DMs or has been deleted. It's (content includes '{}' with reminderId '{}'). The reason for (failure '{}')""",
-                        reminder.pendingReminders().getAuthorId(),
-                        reminder.pendingReminders().getContent(),
-                        reminder.pendingReminders().getId(), failure);
+                        user has disabled DMs or has been deleted. It's reminderId '{}' content includes '{}'.""",
+                        pendingReminder.getAuthorId(), pendingReminder.getContent(),
+                        pendingReminder.getId(), failure);
             return;
 
         }
 
-        int failureAttempts = reminder.pendingReminders().getFailureAttempts() + 1;
-        Instant remindAt = Instant.now().plusSeconds(60);
+        int failureAttempts = pendingReminder.getFailureAttempts() + 1;
+        Instant remindAt = Instant.now().plus(1, ChronoUnit.MINUTES);
         database.write(context -> context.newRecord(PENDING_REMINDERS)
-            .setId(reminder.pendingReminders().getId())
-            .setCreatedAt(Instant.from(reminder.pendingReminders().getCreatedAt()))
-            .setGuildId(reminder.pendingReminders().getGuildId())
-            .setChannelId(reminder.pendingReminders().getChannelId())
-            .setAuthorId(reminder.pendingReminders().getAuthorId())
+            .setId(pendingReminder.getId())
+            .setCreatedAt(Instant.from(pendingReminder.getCreatedAt()))
+            .setGuildId(pendingReminder.getGuildId())
+            .setChannelId(pendingReminder.getChannelId())
+            .setAuthorId(pendingReminder.getAuthorId())
             .setRemindAt(remindAt)
-            .setContent(reminder.pendingReminders().getContent())
+            .setContent(pendingReminder.getContent())
             .setFailureAttempts(failureAttempts)
             .insert());
```

Comment:
Could you try the following instead:

```java
pendingReminder.setRemindAt(remindAt);
pendingReminder.setFailureAttempts(failureAttempts);
pendingReminder.insert();
```

maybe that works as well, then you could get rid of everything else here

Reply:
that should work and definitely would clean up the code, ill look into it.

Reply:
updated as requested and pushed the local repo